xmtp_mls/src/groups/mls_sync.rs

```diff
@@ -1254,18 +1259,114 @@ where
+                    {
+                        tracing::error!("Failed to restore group membership: {}", e);
+                    }
+                }
+
+                // Handle addition to the pending remove list
+                if metadata_info
+                    .pending_remove_added
+                    .iter()
+                    .any(|id| id.inbox_id == current_inbox_id)
+                {
+                    if let Err(e) = storage.db().update_group_membership(
+                        &self.group_id,
+                        GroupMembershipState::PendingRemove,
+                    ) {
+                        // todo(mch):should we return a retryable error here?
+                        tracing::error!(
+                            "Failed to update group membership to PendingRemove: {}",
+                            e
+                        );
+                    }
+                }
+            }
+        }
+
+        // Process admin actions based on current group state
+        // If the current user is admin/super-admin and there are pending remove requests, mark the group accordingly
+        // todo: we need to check if other clients and the same time one of the admins wants to leave the group,
+        // in this case, both the admin and other clients are in the pending remove list.
+        // but still, the admin client needs to react to others and remove them from the pending remove list.
+        match extract_group_mutable_metadata(mls_group) {
+            Ok(metadata) => {
+                let pending_remove = &metadata.pending_remove_list;
+                let is_admin = metadata.admin_list.contains(&current_inbox_id)
+                    || metadata.super_admin_list.contains(&current_inbox_id);
+                let has_pending_removes = !pending_remove.is_empty();
+                let current_user_not_pending = !pending_remove.contains(&current_inbox_id);
+
+                // Update group status if admin needs to handle pending removes
+                if is_admin && current_user_not_pending && has_pending_removes {
+                    // Update the group's pending leave request status
+                    // if let Err(e) = storage
+                    //     .db()
+                    //     .set_group_has_pending_leave_request_status(&self.group_id, Some(true))
+                    // {
+                    //     //     //     .map_err(|e| {
+                    //     tracing::error!("Failed to set group pending leave request status: {}", e);
+                    // } else {
+                    //     //     //         IntentError::Storage(e.into())
+                    //     //     //     })?;
+                    //     //
+                    //     tracing::info!("Marked the group as having pending leave requests");
+                    // }
+                }
+            }
+            Err(GroupMutableMetadataError::MissingExtension) => {
+                //todo(mch): retryable_error?
```

Comment:
question: in this case, do we need to make the error retryable? fyi: all mutable metadata errors don't have retryable error implementation.